hydra fails to build on release-17.03

I noticed that hydra fails to build on the latest `release-17.03` (8d9c8383f921e84eac21d1125cbaee1f3fa97aed):

```
$ nix-build -A hydra
...
Making all in hydra-eval-jobs
make[3]: Entering directory '/tmp/nix-build-hydra-2017-06-22.drv-0/hydra-803833aba77e1082c14857aa26933fc7fe5ae190-src/src/hydra-eval-jobs'
g++ -DHAVE_CONFIG_H -I. -I../..    -I/nix/store/avm18zin1yx7h93gw8lq0kazlwl0ip28-boehm-gc-7.6.0-dev/include -I/nix/store/gx58k0g0z3vcqcw9wbg8hdv79rnddhd3-nix-1.12pre5511_c94f3d55-dev/include/nix -g -O2 -std=c++17 -include nix/config.h -c -o hydra-eval-jobs.o hydra-eval-jobs.cc
hydra-eval-jobs.cc: In function 'void findJobsWrapped(nix::EvalState&, nix::JSONObject&, const AutoArgs&, nix::Value&, const string&)':
hydra-eval-jobs.cc:111:47: error: cannot convert 'nix::DrvInfo' to 'bool' for argument '3' to 'std::experimental::fundamentals_v1::optional<nix::DrvInfo> nix::getDerivation(nix::EvalState&, nix::Value&, bool)'
         if (getDerivation(state, v, drv, false)) {
                                               ^
hydra-eval-jobs.cc:116:21: error: 'std::__cxx11::string nix::DrvInfo::system' is private within this context
             if (drv.system == "unknown")
                     ^~~~~~
In file included from hydra-eval-jobs.cc:12:0:
/nix/store/gx58k0g0z3vcqcw9wbg8hdv79rnddhd3-nix-1.12pre5511_c94f3d55-dev/include/nix/get-drvs.hh:21:20: note: declared private here
     mutable string system;
                    ^~~~~~
hydra-eval-jobs.cc:121:37: error: 'std::__cxx11::string nix::DrvInfo::name' is private within this context
             res.attr("nixName", drv.name);
                                     ^~~~
In file included from hydra-eval-jobs.cc:12:0:
/nix/store/gx58k0g0z3vcqcw9wbg8hdv79rnddhd3-nix-1.12pre5511_c94f3d55-dev/include/nix/get-drvs.hh:20:20: note: declared private here
     mutable string name;
                    ^~~~
hydra-eval-jobs.cc:122:36: error: 'std::__cxx11::string nix::DrvInfo::system' is private within this context
             res.attr("system", drv.system);
                                    ^~~~~~
In file included from hydra-eval-jobs.cc:12:0:
/nix/store/gx58k0g0z3vcqcw9wbg8hdv79rnddhd3-nix-1.12pre5511_c94f3d55-dev/include/nix/get-drvs.hh:21:20: note: declared private here
     mutable string system;
                    ^~~~~~
```

@edolstra, @zimbatm I guess this is related to the recent `nixUnstable` [upgrade](https://github.com/NixOS/nixpkgs/commit/02cecb3e8da67fb619bbd45d19296d7ddd44c227).

--- a/pkgs/development/tools/misc/hydra/default.nix
+++ b/pkgs/development/tools/misc/hydra/default.nix
@@ -62,15 +62,15 @@ let
   };
 in releaseTools.nixBuild rec {
   name = "hydra-${version}";
-  version = "2017-06-22";
+  version = "2017-07-24";
 
   inherit stdenv;
 
   src = fetchFromGitHub {
     owner = "NixOS";
     repo = "hydra";
-    rev = "803833aba77e1082c14857aa26933fc7fe5ae190";
-    sha256 = "1cnxpsan8l6fnbr73n0qxxq1szlda1n3qfkk9k9ic8ijk7md4pvs";
+    rev = "a6d9201947aa1468d31ef5c2651251ceeefceb5c";
+    sha256 = "0hk5pxzn94ip3nyccxl91zc5n6wd1h2zcbhdq9p38wa4lrnnm5zv";
   };
 
   buildInputs =

